### PR TITLE
[Procheta | Japneet] Refactor. testcases and catalog

### PR DESCRIPTION
**Pull request by prochetab1999**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -11,6 +12,7 @@ const App = () => (
         <Route path="/" element={<SearchScreen />} />
         <Route path="/search" element={<SearchResult />} />
         <Route path="/invoice" element={<InvoiceScreen />} />
+        <Route path="/quote" element={<SelectJourney />} />
```
[the remark this answers is not in the document]
Comment (prochetab1999) [reply src/App.js]: updated

File: src/components/Quote.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.quote-vehicle {
```
[the remark this answers is not in the document]
Comment (prochetab1999) [reply src/components/Quote.css]: updated

File: .env [hunk 3]
```diff
@@ -1 +1,5 @@
-REACT_APP_GOOGLE_MAPS_API_KEY = ''
\ No newline at end of file
+REACT_APP_GOOGLE_MAPS_API_KEY = ''
+REACT_APP_BUYER_APP_ID=''
```
Comment (hariharanweb) [inline .env:2]: oh this should not be here. this is in BAP

File: src/components/Catalog.spec.js [hunk 4]
```diff
@@ -6,7 +6,19 @@ import '@testing-library/jest-dom';
 jest.mock('./Item', () => () => <div data-testid="Item" />);
 
 let catalog = {};
+let bookingInformation = {};
+let bookingResponse = {};
+let showModal;
+let loadingJourney;
+let handleClose;
+let onSelectJourney;
 describe('Basic functionality', () => {
```
Comment (hariharanweb) [inline src/components/Catalog.spec.js:15]: Name the testcase as Catalog

File: src/components/Quote.css [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.select-vehicle {
+  display: flex;
+  justify-content: center;
+}
+.quote-vehicleImg{
```
Comment (hariharanweb) [inline src/components/Quote.css:5]: rename this too?

File: src/screens/SearchResult.js [hunk 6]
```diff
@@ -1,82 +1,66 @@
...
+            {
+              id: searchResults[0]?.bpp_providers[0]?.items[0]?.id,
+              fulfillment_id:
+                searchResults[0]?.bpp_providers[0]?.items[0]?.fulfillment_id,
+              descriptor:
+                searchResults[0]?.bpp_providers[0]?.items[0]?.descriptor,
+              price: searchResults[0]?.bpp_providers[0]?.items[0]?.price,
+              category_id:
+                searchResults[0]?.bpp_providers[0]?.items[0]?.category_id,
+              tags: searchResults[0]?.bpp_providers[0]?.items[0]?.tags,
```
Comment (hariharanweb) [inline src/screens/SearchResult.js:53]: tags are not standard. Lets remove that

File: src/utilities/ContextBuilder.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import moment from 'moment';
+import { v4 as uuid } from 'uuid';
+
+const getContext = (action) => ({
```
Comment (hariharanweb) [inline src/utilities/ContextBuilder.js:4]: Context should not be built in frontend. Its in BAP

Comment (prochetab1999) [reply src/utilities/ContextBuilder.js:4]: In Search flow, it interacts with the Gateway. But in this scenario we have to send it directly from BAP to BPP

File: .env (hunk 3 above)
Comment (prochetab1999) [reply .env:2]: updated

File: src/components/Catalog.spec.js (hunk 4 above)
Comment (prochetab1999) [reply src/components/Catalog.spec.js:15]: Updated for this file and other files as well

File: src/components/Quote.css (hunk 5 above)
Comment (prochetab1999) [reply src/components/Quote.css:5]: Updated

File: src/screens/SearchResult.js (hunk 6 above)
Comment (prochetab1999) [reply src/screens/SearchResult.js]: Updated

File: .env [hunk 8]
```diff
@@ -1 +1,3 @@
-REACT_APP_GOOGLE_MAPS_API_KEY = ''
\ No newline at end of file
+REACT_APP_GOOGLE_MAPS_API_KEY = ''
+REACT_APP_SELLER_APP_ID=''
```
Comment (hariharanweb) [inline .env:2]: Still here ?

File: src/utilities/ContextBuilder.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  action,
+  transaction_id: uuid(),
+  timestamp: moment().format(),
```
Comment (hariharanweb) [inline src/utilities/ContextBuilder.js:7]: This can also be moved;. So you can remove moment js from dependencies.

File: .env (hunk 8 above)
Comment (prochetab1999) [reply .env:2]: Seller App ID is needed for BPP context builder in Frontend for /select

Comment (japneetbhatia) [reply .env]: Updated

**Pull request merged**